modules/live-schedule-impl/src/main/java/org/opencastproject/liveschedule/impl/LiveScheduleServiceImpl.java

```diff
@@ -679,23 +677,19 @@ MediaPackage replaceAndDistributeAcl(MediaPackage previousMp, AccessControlList
   void addLivePublicationChannel(Organization currentOrg, MediaPackage mp) throws LiveScheduleException {
     logger.debug("Adding live channel publication element to media package {}", mp);
     String engageUrlString = null;
-    String playerPath = null;
     if (currentOrg != null) {
       engageUrlString = StringUtils.trimToNull(currentOrg.getProperties().get(ENGAGE_URL_PROPERTY));
-      playerPath = StringUtils.trimToNull(currentOrg.getProperties().get(PLAYER_PROPERTY));
     }
     if (engageUrlString == null) {
       engageUrlString = serverUrl;
       logger.info(
               "Using 'server.url' as a fallback for the non-existing organization level key '{}' for the publication url",
               ENGAGE_URL_PROPERTY);
     }
-    if (playerPath == null)
-      playerPath = DEFAULT_PLAYER_PATH;
 
     try {
       // Create new distribution element
-      URI engageUri = URIUtils.resolve(new URI(engageUrlString), playerPath + "?id=" + mp.getIdentifier().compact());
+      URI engageUri = URIUtils.resolve(new URI(engageUrlString), PLAYER_PATH + mp.getIdentifier().compact());
```

Comment:
This code, building the link to the watch page, is still duplicated. We might want to use this opportunity to factor this into some kind of helper method, so that every module that needs a link to a player page can just say `whateverService.getPlayerLink(mpId)`. :thinking:

Reply:
+1

Reply:
Moving the redirect-path creation path to a separate common service will also make it easier for a site, like us, that uses 2 different engage path's for one organization during one workflow (the first published path forces a re-direct to Engage login before re-redirecting to the player, the second publish path doesn't) to encapsulate our custom[...]ness into one isolated service.

Reply:
I've looked into this and I agree that if these URLs are an integral part of the organization, it might make sense to add methods like `getAdminUrI()` or `getPlayerLink(…)` to the organization at which point it would also make sense to move the configuration for those from the general, optional properties to dedicated configuration values.

But this is somewhat unrelated to this since it wouldn't make much sense to do this only with the engage URL and another, likely larger patch which I'm happy to review if one of you wants to provide the code ;-P